framework/finalizer.go

```diff
@@ -117,8 +120,10 @@ func (f *Framework) removeFinalizer(ctx context.Context, obj interface{}) error
 		return microerror.Mask(err)
 	}
 	operation := func() error {
-		res := f.restClient.Patch(types.JSONPatchType).AbsPath(path).Body(p).Do()
-		if res.Error() != nil {
+		res := restClient.Patch(types.JSONPatchType).AbsPath(path).Body(p).Do()
+		if errors.IsNotFound(res.Error()) {
+			return nil // the object is already gone, nothing to do.
```

Comment:
I do not really follow here. There is a CR which has a finalizer. When it receives a delete event we want to remove the finalizer. So we remove the finalizer, get a duplicated event and then this here happens right? Does the duplicated event still contain the CR including the finalizer we actually removed? If this is the case we might want to check our informer implementation. Please create an issue for this. I will check on this once you got the other integration test we talked about implemented. Then I have an easy base to debug this properly.

Reply:
Yes, this is the case here. The object contains the CR we removed and still has the finalizer, the only change that I could find was the `resourceVersion` and some timestamps. I had the feeling that this might be related to the `DeletionGracePeriod` which is set to 0 seconds, but I couldn't really find something in the k8s docs to confirm this.

Reply:
Ok so when the resource version is different then we cannot have a problem in our informer because the object seems to be a different one. Anyway, would be good to reach out to upstream once we have the other integration test. Thanks.

Reply:
I created an issue here for now: https://github.com/giantswarm/giantswarm/issues/2897